src/transforms/geoip.rs

```diff
@@ -0,0 +1,256 @@
+use super::Transform;
+
+use crate::{
+    event::Event,
+    topology::config::{DataType, TransformConfig},
+};
+use serde::{Deserialize, Serialize};
+use string_cache::DefaultAtom as Atom;
+
+use indexmap::IndexMap;
+use std::net::IpAddr;
+use std::str::FromStr;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct GeoipConfig {
+    pub source: Atom,
+    pub database: String,
+    #[serde(default = "default_geoip_target_field")]
+    pub target: String,
+}
+
+pub struct Geoip {
+    pub dbreader: maxminddb::Reader<Vec<u8>>,
+    pub source: Atom,
+    pub target: String,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct GeoipDecodedData {
+    pub city_name: String,
+    pub continent_code: String,
+    pub country_code: String,
+    pub time_zone: String,
+    pub latitude: String,
+    pub longitude: String,
+    pub postal_code: String,
+}
+
+fn default_geoip_target_field() -> String {
+    "geoip".to_string()
+}
+
+#[typetag::serde(name = "geoip")]
+impl TransformConfig for GeoipConfig {
+    fn build(&self) -> Result<Box<dyn Transform>, crate::Error> {
+        let reader = maxminddb::Reader::open_readfile(self.database.clone()).unwrap();
```

Comment:
Instead of unwrapping here, we should propogate the error so that it gets displayed to users. That way they'll know if they messed up the configuration or something like that.